### PR TITLE
add GitHub auth support

### PR DESCRIPTION
**Pull request by minrk**
- [x] support client_id, client_secret in github provider
- [x] better error when rate limit exceeded
- [x] support env in helm chart

Once this lands, we can add the secrets for https://github.com/jupyterhub/mybinder.org-deploy/issues/58

### CODE REVIEW

File: helm-chart/binderhub/templates/deployment.yaml [hunk 1]
```diff
@@ -62,6 +62,27 @@ spec:
             secretKeyRef:
               name: binder-secret
               key: "binder.hub-token"
+        {{ if .Values.github.clientId -}}
```
Comment (yuvipanda) [inline helm-chart/binderhub/templates/deployment.yaml]: While this is ok for now, I think in the long run we should load the secret as a volume and read these from binderhub_config.py. Just  to keep the deployment definition clean.

**Comment by yuvipanda** [comment]
@minrk I'm curious about possible merge strategies here. How coupled should merges here to deployment be? In past lives, I've found it very important that merges in repos like this are immediately deployed - this makes sure that we don't end up with a large number of changes that are undeployed, potentially breaking on the next person who does a deploy. If we do that, then a possible CR workflow is: 

1. someone else does an 'approve'
2. person who did the PR merges
3. person who did the PR immediately deploys / rolls back as appropriate.

what do you think?

**Comment by minrk** [comment]
After chatting a bit: I think greenkeeper-style automatic PRs to staging when PRs land here or repo2docker would be pretty nice to keep the size of deploys small, but stopping short of automatic deployment of PRs from here to staging for now.

**Comment by minrk** [comment]
rate limit prometheus PR coming soon!

**Pull request merged**